crates/control/src/motion/motor_command_optimizer.rs

```diff
@@ -0,0 +1,128 @@
+use color_eyre::Result;
+use context_attribute::context;
+use framework::{AdditionalOutput, MainOutput};
+use types::{
+    joints::{arm::ArmJoints, Joints},
+    motion_selection::MotionSelection,
+    motor_commands::MotorCommand,
+    parameters::MotorCommandOptimizerParameters,
+    sensor_data::SensorData,
+};
+
+pub struct MotorCommandOptimizer {
+    position_offset: Joints<f32>,
+    state: State,
+}
+
+pub enum State {
+    Optimizing,
+    Resetting,
+}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    pub motor_commands: Input<MotorCommand<f32>, "motor_commands">,
+    pub sensor_data: Input<SensorData, "sensor_data">,
+    pub motion_selection: Input<MotionSelection, "motion_selection">,
+
+    pub parameters:
+        Parameter<MotorCommandOptimizerParameters, "motor_command_optimizer_parameters">,
```

Comment:
please specify the parameters individually. You do not need this struct

Reply:
Done